### PR TITLE
Kore AST verifier.

### PR DESCRIPTION
**Pull request by virgil-serbanuta**

### CODE REVIEW

File: src/main/haskell/language-kore/src/Data/Kore/ASTVerifier/AttributesVerifier.hs [hunk 1]
```diff
@@ -4,7 +4,8 @@ import           Data.Kore.AST
 import           Data.Kore.ASTVerifier.Error
 import           Data.Kore.ASTVerifier.PatternVerifier
 import           Data.Kore.Error
-import qualified Data.Set as Set
+import           Data.Kore.IndexedModule.IndexedModule
```
Comment (traiansf) [inline src/main/haskell/language-kore/src/Data/Kore/ASTVerifier/AttributesVerifier.hs]: is this needed?

File: src/main/haskell/language-kore/src/Data/Kore/ASTVerifier/ModuleVerifier.hs [hunk 2]
```diff
@@ -1,62 +1,39 @@
-module Data.Kore.ASTVerifier.ModuleVerifier (verifyModule,
-                                             verifyUniqueNames) where
+module Data.Kore.ASTVerifier.ModuleVerifier
+    (verifyModule, verifyUniqueNames) where
```
Comment (traiansf) [inline src/main/haskell/language-kore/src/Data/Kore/ASTVerifier/ModuleVerifier.hs]: any reason to not follow existing conventions for indenting this?

File: src/main/haskell/language-kore/src/Data/Kore/ASTVerifier/AttributesVerifier.hs (hunk 1 above)
Comment (virgil-serbanuta) [reply src/main/haskell/language-kore/src/Data/Kore/ASTVerifier/AttributesVerifier.hs]: Yes

File: src/main/haskell/language-kore/src/Data/Kore/ASTVerifier/ModuleVerifier.hs (hunk 2 above)
Comment (virgil-serbanuta) [reply src/main/haskell/language-kore/src/Data/Kore/ASTVerifier/ModuleVerifier.hs]: None, sorry.

File: src/main/haskell/language-kore/src/Data/Kore/IndexedModule/IndexedModule.hs [hunk 3]
```diff
@@ -14,13 +33,29 @@ import           Control.Monad   (foldM)
...
+IDs.
+
+It contains mappings from IDs to the sentence that declares them
+(or to the 'SortDescription' for sorts). Note that only symbols defined
+in the current module are included.
+
+It also contains the imported modules as 'IndexedModule's and all the other
+module data in raw-ish form.
+
+All 'IndexedModule' instances should either be returned by '' or they should
```
Comment (traiansf) [inline src/main/haskell/language-kore/src/Data/Kore/IndexedModule/IndexedModule.hs]: returned by who?

**Pull request merged**